docs/migrations-guidelines.md

```diff
@@ -0,0 +1,89 @@
+# Migration and Testing Guidelines
+
+## Overview
+
+This document outlines best practices and guidelines for writing migration scripts and their corresponding test cases.
+
+The focus is on ensuring data integrity, handling errors gracefully, and maintaining consistency across mobile app versions.
+
+We always look for improvement, if you see anything that could be improved, please open a PR against this guidelines.
+
+You can also check an example of a migration on MetaMask mobile app [here](https://github.com/MetaMask/metamask-mobile/blob/1855bd674e33bb0ece06fb6d8f09a4e5df46a108/app/store/migrations/044.ts#L1)
+
+## Migration Guidelines
+
+1. **Pre-Validation Checks**:
+
+- Ensure the state to be migrated is valid before running the migration logic,
+- Verify the version and integrity of the state with utility functions like `ensureValidState`.
+
+2. **Error Handling**:
+
+The following prevents manipulation of potentially corrupted data, enforcing corrective action before proceeding.
+
+- Log erros with `captureException` from `@sentry/react-native`, which is crucial for diagnosing issues post-migration,
+- Ensure that error messages are descriptive: include the migration number and a clear description of the issue,
+- If an exception is detected, indicating potential data corruption, halt the migration process and return the intial state,
+
+3. **State Integrity Checks**:
+
+- Checks the state's structure and types carefully,
+- Validate the state and its nested properties using functions like `isObject` and `hasProperty`,
+- Prevent data corruption by halting the migration on any inconsistencies and logging errors.
+
+5. **Avoid Type Casting**:
```

Comment:
Avoiding type casting is a general rule we have for all TypeScript. Maybe we can omit this because it's a rule for all code.

The migrations do involve more type checking than usual, but perhaps we can explain that in the "State integrity checks" section. I would explain that the input state has type `unknown` because that's how we use TypeScript to validate state integrity before proceeding with a migration.